media/system/js/treeselectmenu.js

```diff
@@ -2,117 +2,193 @@
-				hidden++;
-			}
-			else {
-				$(this).show();
-			}
-		});
-		if(hidden == $list_elements.length)
-		{
-			$("#noresultsfound").show();
-		}
-	});
-
-	// Checks all checkboxes the tree
-	$('#treeCheckAll').click(function()
-	{
-		$('.treeselect input').attr('checked', 'checked');
-	});
-
-	// Unchecks all checkboxes the tree
-	$('#treeUncheckAll').click(function()
-	{
-		$('.treeselect input').attr('checked', false);
-	});
-
-	// Checks all checkboxes the tree
-	$('#treeExpandAll').click(function()
-	{
-		$('ul.treeselect ul.treeselect-sub').show();
-		$('ul.treeselect i.treeselect-toggle').removeClass('fa-chevron-right').addClass('fa-chevron-down');
-	});
-
-	// Unchecks all checkboxes the tree
-	$('#treeCollapseAll').click(function()
-	{
-		$('ul.treeselect ul.treeselect-sub').hide();
-		$('ul.treeselect i.treeselect-toggle').removeClass('fa-chevron-down').addClass('fa-chevron-right');
-	});
-	// Take care of children check/uncheck all
-	$('a.checkall').click(function()
-	{
-		$(this).parents().eq(5).find('ul.treeselect-sub input').attr('checked', 'checked');
-	});
-	$('a.uncheckall').click(function()
-	{
-		$(this).parents().eq(5).find('ul.treeselect-sub input').attr('checked', false);
-	});
-
-	// Take care of children toggle all
-	$('a.expandall').click(function()
-	{
-		var $parent = $(this).parents().eq(6);
-		$parent.find('ul.treeselect-sub').show();
-		$parent.find('ul.treeselect-sub i.treeselect-toggle').removeClass('fa-chevron-right').addClass('fa-chevron-down');
-	});
-	$('a.collapseall').click(function()
-	{
-		var $parent = $(this).parents().eq(6);
-		$parent.find('li ul.treeselect-sub').hide();
-		$parent.find('li i.treeselect-toggle').removeClass('fa-chevron-down').addClass('fa-chevron-right');
-	});
-});
+Joomla = window.Joomla || {};
+
+(function (Joomla) {
+    document.addEventListener('DOMContentLoaded', function () {
+        var treeselectmenu = document.querySelector("div#treeselectmenu").innerHTML;
+        var treeselect = document.querySelectorAll('.treeselect li');
+
+        var li, div, i;
+        for (i = 0; i < treeselect.length; i++) {
+            li = event.target;
+            div = li.querySelector('div.treeselect-item');
+
+            // Add icons
+            treeselect[i].insertAdjacentHTML("beforebegin", "<span class='icon-'></span>");
+
+            if (li.querySelectorAll('ul.treeselect-sub').length) {
+                // Add classes to Expand/Collapse icons
+                li.querySelector('span.icon-').classList.add('treeselect-toggle');
+                li.querySelector('span.icon-').classList.add('fa-chevron-down');
+
+                // Append drop down menu in nodes
+                var element = div.querySelector('label');
+                element.insertAdjacentHTML('afterend', treeselectmenu);
+
+                if (!li.querySelectorAll('ul.treeselect-sub ul.treeselect-sub').length) {
+                    li.querySelector('div.treeselect-menu-expand').classList.remove();
```

Comment:
In the original code, it removes the element and in the refactored code, it calls classList.remove(), Couldn't relate a connection. Did you check running the code and testing the functionality?